backend/src/v4/services/api.js

```diff
@@ -98,7 +98,7 @@ APIService.createAppAsync = async (config, v5Init = true) => {
 	const app = express();
 	app.use(cookieParser());
 
-	if (config && !config.using_ssl && config.public_protocol === "https") {
+	if (config && ((!config.using_ssl && config.public_protocol === "https") || config.trust_proxy)) {
```

Comment:
why was this added? 😮

Reply:
because I have to have trust proxy set to true to be able to test it in the e2e tests, otherwise x-forwarded-for header is ignored and the ip is always the same

![image](https://github.com/user-attachments/assets/43301e10-2214-47ce-b3a8-01fb2104ee5d)

Reply:
i don't think you need a e2e test for this, just make sure the unit test works it's fine.

To allow trust proxy when it's a http protocol is a bit weird.